connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -100,20 +102,39 @@ public Worker(
         this.config = config;
         this.workerMetricsGroup = new WorkerMetricsGroup(metrics);
 
-        // Internal converters are required properties, thus getClass won't return null.
-        this.internalKeyConverter = plugins.newConverter(
-                config.getClass(WorkerConfig.INTERNAL_KEY_CONVERTER_CLASS_CONFIG).getName(),
-                config
-        );
+        // Internal key/value converters are currently deprecated, instead JsonConverter is used by default, refer KIP-174.
+        // Currently Internal converters are required properties, thus getClass won't return null.
+
+        boolean isJsonConverterKeyClassUsed = config.getClass(WorkerConfig.INTERNAL_KEY_CONVERTER_CLASS_CONFIG).getName().equals("org.apache.kafka.connect.json.JsonConverter");
+        boolean isJsonConverterValueClassUsed = config.getClass(WorkerConfig.INTERNAL_VALUE_CONVERTER_CLASS_CONFIG).getName().equals("org.apache.kafka.connect.json.JsonConverter");
+
+        if (!isJsonConverterKeyClassUsed && !isJsonConverterValueClassUsed) {
+            this.internalKeyConverter =
```

Comment:
Doesn't the KIP say we're going to log a warning if the setting is specified? I don't see that anywhere, just the configs marked deprecated.

Reply:
Ah, sorry I missed to log that. I'll log the warning if these configurations value is set by user explicitly.